tutorials/klusterai-api/rag.ipynb

```diff
@@ -0,0 +1,846 @@
+    "from llama_index.core import SimpleDirectoryReader\n",
+    "\n",
+    "# Load documents from the PDF file\n",
+    "print(f\"Loading PDF from {pdf_dir}...\")\n",
+    "pdf_reader = SimpleDirectoryReader(input_dir=pdf_dir)\n",
+    "documents = pdf_reader.load_data()\n",
+    "\n",
+    "print(f\"Loaded {len(documents)} document(s) from PDF file\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df42ac97",
+   "metadata": {},
+   "source": [
+    "## Setup RAG\n",
+    "\n",
+    "If the entire PDF document is sent to the LLM, it would consume the entire context window with no success.\n",
+    "\n",
+    "To address this, the document needs to be divided into smaller pieces called chunks. <a href=\"https://docs.llamaindex.ai/en/stable/\" target=\"_blank\">LlamaIndex</a> provides an efficient way to accomplish this.\n",
+    "\n",
+    "- **LLM**: the model responsible for generating the responses from the knowledge base.\n",
+    "- **Embedding model**: the model used to convert text into vectors (embeddings), enabling semantic similarity searches.\n",
+    "- **Chunking parameters**: defines how documents are split into smaller chunks (nodes) for indexing:\n",
+    "    - `chunk_size`: the size of each chunk (in tokens).\n",
+    "    - `chunk_overlap`: the number of tokens overlapping between consecutive chunks.\n",
+    "\n",
+    "To set up with **kluster.ai**, configure `OpenAILike` for the LLM and `OpenAILikeEmbedding` for the embedding model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "52d3b654",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "kluster.ai embedding model configured.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from llama_index.llms.openai_like import OpenAILike\n",
+    "from llama_index.embeddings.openai_like import OpenAILikeEmbedding\n",
+    "from llama_index.core import Settings\n",
+    "\n",
+    "# Set up the llamaIndex client with kluster.ai\n",
+    "llm = OpenAILike(\n",
+    "    model=\"meta-llama/Llama-4-Maverick-17B-128E-Instruct-FP8\",\n",
+    "    api_base=base_url,\n",
+    "    api_key=api_key,\n",
+    "    is_chat_model=True\n",
+    ")\n",
+    "\n",
+    "# Set up the embedding model\n",
+    "embed_model = OpenAILikeEmbedding(\n",
+    "    model_name=\"BAAI/bge-m3\",\n",
+    "    api_base=base_url,\n",
+    "    api_key=api_key\n",
+    ")\n",
+    "\n",
+    "# Set the global settings for LlamaIndex\n",
+    "Settings.llm = llm\n",
+    "Settings.embed_model = embed_model\n",
+    "Settings.chunk_size = 512  # Set chunk size for document splitting\n",
+    "Settings.chunk_overlap = 20  # Set chunk overlap for document splitting\n",
+    "\n",
+    "print(\"kluster.ai embedding model configured.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c9cc04f1",
+   "metadata": {},
+   "source": [
+    "### Create the VectorStore\n",
```

Comment:
```suggestion
    "### Create the vector store\n",
```

I try to avoid using specific code references in titles. It can be done, but when we can avoid it, let's do it